### PR TITLE
Remove `cra-ts-essentials`

### PR DESCRIPTION
**Pull request by tmeasday**
## What I did

- Moved `composeStories` tests from `cra-ts-essentials` to `renderers/react` (we may want to discuss)
- Remove example

## How to test

CI

### CODE REVIEW

File: code/renderers/react/src/docs/__testfixtures__/8140-js-prop-types-oneof/docgen.snapshot [hunk 1]
```diff
@@ -46,14 +47,14 @@ Alert.__docgenInfo = {
           \\"computed\\": false
         }]
       },
-      \\"required\\": false,
-      \\"description\\": \\"\\"
+      \\"required\\": false
```
[the remark this answers is not in the document]
Comment (tmeasday) [reply code/renderers/react/src/docs/__testfixtures__/8140-js-prop-types-oneof/docgen.snapshot:50]: Possibly a result of the `yarn.lock` changes.

File: code/renderers/react/src/__test__/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## Integration tests for @storybook/react.
+
+As these tests are implemented as stories, we may consider moving them inside the `template/stories` folder.
```
[the remark this answers is not in the document]
Comment (tmeasday) [reply code/renderers/react/src/__test__/README.md:3]: Let's put this in a `test` folder

File: code/renderers/react/src/docs/__testfixtures__/8140-js-prop-types-oneof/docgen.snapshot (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmeasday) [reply code/renderers/react/src/docs/__testfixtures__/8140-js-prop-types-oneof/docgen.snapshot:50]: Probably, so let's ignore.

File: code/renderers/react/src/__test__/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tmeasday) [reply code/renderers/react/src/__test__/README.md:3]: @shilman so the issue with doing that is that these stories/tests are written in TS and actually make heavy use of TS which we probably don't want to lose. But we aren't allowed to use TS stories in the renderers `template/stories` folder for technical reasons... 🤔

[the remark this answers is not in the document]
Comment (shilman) [reply code/renderers/react/src/__test__/README.md:3]: Hmm what a pain. We could add the stories to`/template/stories` but add the tests to `/tests`?

[the remark this answers is not in the document]
Comment (tmeasday) [reply code/renderers/react/src/__test__/README.md:3]: @shilman the tests can be TS as they aren't used by the SB, but they make use of the types exported by the component/stories so we are kind of stuck.

[the remark this answers is not in the document]
Comment (shilman) [reply code/renderers/react/src/__test__/README.md:3]: @tmeasday We're loading the stories with `esbuild-loader` so technically we could use TS there in a pinch

[the remark this answers is not in the document]
Comment (tmeasday) [reply code/renderers/react/src/__test__/README.md:3]: We don't load the stories from the renderer with the esbuild loader as they rely on renderer specific config (like JSX) that the esbuild loader won't always support

[the remark this answers is not in the document]
Comment (tmeasday) [reply code/renderers/react/src/__test__/README.md:3]: I think let's leave it for now and resolve this at a later state.

**Review by shilman — APPROVED** [review]
LGTM! Still don't like the `__test__` convention, but whatever

**Pull request merged**